### PR TITLE
Tests related to GobPie configuration

### PR DESCRIPTION
**Pull request by karoliineh**
- [x] reading example conf (from readMe) works
- [x] reading complete conf (has every field possible) works
   - combinations of different boolean values (optional)
- [x] when reading non-complete conf (only required values), the optional configuration default values are right
- [x] error test: GobPie configuration file missing
- [x] error test: GobPie configuration has some field that is not supposed to be there
- [x] error test: GobPie configuration has wrong JSON syntax
- [x] error test: Gobpie configuration has a required field missing
   - [x] one per each required field

### CODE REVIEW

File: src/main/java/gobpie/GobPieConfReader.java [hunk 1]
```diff
@@ -28,7 +28,9 @@ public class GobPieConfReader {
 
     private final MagpieServer magpieServer;
     private final String gobPieConfFileName;
-    private static final Logger log = LogManager.getLogger(GobPieConfReader.class);
+    //private static final Logger log = LogManager.getLogger(GobPieConfReader.class);
+    //changed (erased static)
```
Comment (karoliineh) [inline src/main/java/gobpie/GobPieConfReader.java]:
```suggestion
```

File: src/test/java/GobPieConfTest.java [hunk 2]
```diff
@@ -1,59 +1,180 @@
...
 
+    @SystemStub
+    private SystemOut systemOut;
+
+    @Test
+    void testReadGobPieConfiguration() {
+        // Mock everything needed for creating GobPieConfReader
+        MagpieServer magpieServer = mock(MagpieServer.class);
+        String gobPieConfFileName = GobPieConfTest.class.getResource("gobpieTest1.json").getFile();
+        GobPieConfReader gobPieConfReader = new GobPieConfReader(magpieServer, gobPieConfFileName);
```
Comment (karoliineh) [inline src/test/java/GobPieConfTest.java]: Extract this block into a separate method that takes the resource name as an argument and use it in all tests.

File: src/test/java/GobPieConfTest.java [hunk 3]
```diff
@@ -1,59 +1,180 @@
...
 
+        assertThrows(GobPieException.class, ()-> {
+            try{
+                GobPieConfiguration actualGobPieConfiguration = gobPieConfReader.readGobPieConfiguration();
+            }
+            catch (JsonSyntaxException e) {
+                assertEquals("GobPie configuration file syntax is wrong.", e.getMessage());
+            }
+        });
+    }
```
Comment (karoliineh) [inline src/test/java/GobPieConfTest.java]:
```suggestion
        GobPieException thrown = assertThrows(GobPieException.class, gobPieConfReader::readGobPieConfiguration);
        assertEquals("GobPie configuration file syntax is wrong.", thrown.getMessage());
```

File: src/test/java/GobPieConfTest.java [hunk 4]
```diff
@@ -1,59 +1,180 @@
...
+        // Mock everything needed for creating GobPieConfReader
+        MagpieServer magpieServer = mock(MagpieServer.class);
+        FileWatcher gobPieConfWatcher = mock(FileWatcher.class);
+        String gobPieConfFileName =  GobPieConfTest.class.getResource("gobpie.json").getFile();
+
+        GobPieConfReader gobPieConfReader = new GobPieConfReader(magpieServer, gobPieConfFileName);
+        doNothing().when(gobPieConfWatcher).waitForModified();
+
+        GobPieConfiguration actualGobPieConfiguration = gobPieConfReader.readGobPieConfiguration();
+        //.waitForModified() juurde jääb kinni
```
Comment (karoliineh) [inline src/test/java/GobPieConfTest.java]: I'll look into this tomorrow.

Comment (karoliineh) [reply src/test/java/GobPieConfTest.java]: @AnetteTaivere I have completed the two error tests now.

File: src/test/java/GobPieConfTest.java (hunk 2 above)
Comment (karoliineh) [reply src/test/java/GobPieConfTest.java]: What is yet left to do in this PR is this refactoring and the `testReadGobPieConfigurationWithExtraField`. For the latter, some message like "There was an unknown option in the GobPie configuration. Please check for any typos." should be communicated to the user. @AnetteTaivere

**Comment by AnetteTaivere** [comment]
The test for configuration with an additional field failed; the expected user message error was not received. @karoliineh

**Comment by karoliineh** [comment]
> The test for configuration with an additional field failed; the expected user message error was not received. @karoliineh

I will soon (hopefully today or tomorrow) look into that and try to fix the issue in GobPie. Thanks.

**Pull request merged**